ui/spinner.go

```diff
@@ -0,0 +1,47 @@
+package ui
+
+import (
+  "time"
+
+  "github.com/briandowns/spinner"
+)
+
+type Spinner struct{
+  spinner *spinner.Spinner
+  text string
+}
+
+func NewSpinner(text string) *Spinner { return defUI.NewSpinner(text) }
+
+func (u *UI) NewSpinner(text string) *Spinner {
+  s := spinner.New(spinner.CharSets[9], 100*time.Millisecond)
+  s.Suffix = " " + text
+  return &Spinner {
+    s,
+    text,
+  }
+}
+
+func (s *Spinner) Start() { defUI.StartSpinner(s) }
+
+func (u *UI) StartSpinner(s *Spinner) {
```

Comment:
Why is this method on the `UI` struct inside of the `Spinner` struct? I think it'd be cleaner/clearer lines of responsibility for the `Spinner` struct to have the values it needs off the `UI` struct copied onto itself (or just a reference to the `UI` struct in general).